core/src/main/java/com/linecorp/armeria/common/http/MulticastHttpRequestFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.http;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.stream.AbstractMulticastStreamFactory;
+import com.linecorp.armeria.common.stream.StreamMessage;
+import com.linecorp.armeria.common.stream.StreamMessageWrapper;
+
+/**
+ * A factory that wraps a {@link HttpRequest} which is not subscribed yet and provides multiple
+ * stream messages that forwards data from the wrapping {@link HttpRequest}.
+ */
+public class MulticastHttpRequestFactory extends AbstractMulticastStreamFactory<HttpObject, HttpRequest> {
+
+    private final HttpHeaders httpHeaders;
+
+    private final boolean httpKeepAlive;
+
+    /**
+     * Creates a new instance wrapping a {@code publisher} and publishing to multiple subscribers.
+     * @param publisher the publisher who will publish data to subscribers
+     */
+    public MulticastHttpRequestFactory(HttpRequest publisher) {
+        super(publisher);
```

Comment:
- Rename publisher to req
- super(req) -> super(requireNonNull(req, "req")); (so that NPE message is not 'publisher'.)